### ISSUE TITLE
🎉🚀 Electron.NET Reloaded: The Adventure Begins! 🚀🎉

### ISSUE THREAD

**Issue by GregorBiswanger** [ElectronNET/Electron.NET#744]
Hey Electron.NET community! 

You might have heard the whispers, but it's time to make it official: "**Electron.NET is dead, long live Electron.NET!**" After a brief hiatus, we're back, and the future looks brighter than ever.

You may have noticed that there haven't been any updates lately - this was mainly due to me being the only active member in the project, while my partner wasn't interested in further development. Add to that numerous pull requests and suboptimal communication, which shook the project's foundations. As a father of three, my private life has also taken up a lot of my time.

But now, everything has changed! 🎆 I'm thrilled to announce that **Florian Rappl** has joined the team, and we're working together with renewed energy. We've updated the project to the latest functional state, with a particular focus on the stability of Websockets. With a brand-new library and a smart architecture, we're now more flexible and future-proof than ever before!

The highlights of our upcoming updates? .NET 6 support, native Electron 20 support, and an impressive stability and performance that shatters all expectations! 💥 But that's just the beginning - we've got many more brilliant features up our sleeves that will make Electron.NET even better. Stay tuned!

And hey, we're only human and appreciate your support! 😉 If you'd like to support us with [donations](https://donorbox.org/electron-net), it would help us show our faces around our better halves. 😄

Get ready for the big update coming in the next few days! 🌟 Together, let's embark on an exciting Electron.NET adventure - stay with us!

Your Electron.NET Team

**Comment by GregorBiswanger** [comment]
Electron.NET 23.6.1 has shipped 🚢

New in this release:
⚡️  Upgraded to native Electron 23
⚡️  Upgraded to .NET 6
🎁  New .NET Socket API (Faster and more stable)
🛠️  Bug fixes and improvements

**Comment by vmeganathan81** [comment]
Thanks, this is great news.

**Comment by xiaoxinxing12** [comment]
谢谢，这是个好消息。

**Comment by kylethomas320** [comment]
New release seems very good so far! 
Startup time has increased by like 10 seconds, which is a huge deal.
Socket communication seems stable for now, although I had to make some minor tweaks for it to work compared to the previous version.

Keep up the good work!

**Comment by GregorBiswanger** [comment]
@kylethomas320 @IgorVolod **Another performance tip for Electron.NET on Windows:** Non-certified Electron desktop applications are also scanned with Windows Defender, which again causes a strong delay. If you set your project directory to be excluded in Windows Defender, the app rips like a rocket. 🚀🚀🚀

**Comment by GregorBiswanger** [comment]
We are still at the beginning and still have numerous performance and stability changes ahead of us. You can definitely stay curious how much power we have ready for you in the future..

**Comment by Islam-alshiki** [comment]
When I `electronize start` my app it stops on the splash screen that `AlwaysOnTop`  
with last version `[23.6.1]`

```bat
ASP.NET Core Application connected... global.electronsocket iun7Rb9ogq_V97uPAAAA 2023-04-06T01:01:17.850Z
ASP.NET Core Application connected... global.electronsocket iun7Rb9ogq_V97uPAAAA 2023-04-06T01:01:19.902Z
ASP.NET Core Application connected... global.electronsocket iun7Rb9ogq_V97uPAAAA 2023-04-06T01:01:19.915Z
Got disconnect! Reason: transport close
Got disconnect! Reason: forced close
```

When I tried [13.5.1], the window appears.

**Comment by tareqimbasher** [comment]
This is great, been waiting for a while for an update. Excited for continued support and improvements!! :rocket:

**Comment by r-pankevicius** [comment]
Dead again?

**Comment by FlorianRappl** [comment]
Not really. Just busy on other projects right now. The focus is always where users are also supportive. Personally, when I see contributions/donations being made that is for me a sign that help is needed / appreciated; so I put the focus there.

**Comment by scottkuhl** [comment]
I hate to say this, but this project is dead.  There has been almost no movement since the big announcement in March.  At a minimum this project needs to keep up with the latest Electron and .NET releases.  :(

**Comment by GregorBiswanger** [comment]
@scottkuhl .NET 8 was only released two weeks ago.. the update will follow.. if you have a feature you want, bring it on...

**Comment by scottkuhl** [comment]
I was just as concerned or more so at the lack of keeping up with Electron versions. This project is on 23 and 28 is about to release.  23 lost support 3 months ago.  I posted in a discussion to create some community guidelines on when the project would update and how anyone could do it to help out, but there was no response.  I don’t see much other response to issues or discussions overall.  That is why I consider this project not active.  Dead was too strong.  Just keeping up with Electron and .NET versions and would go a long way.

**Comment by nepp95** [comment]
> @scottkuhl .NET 8 was only released two weeks ago.. the update will follow.. if you have a feature you want, bring it on...

I think the point is more or less:

Electron.NET is not keeping up to date with Electron or .NET. While you can say, .NET 6 is supported till the end of the year, when your main product is basically a wrapper around both and you have absolutely no support for LTS versions of those, then you are missing a core functionality. And when you do bring it out at the last minute, how are people supposed to have time to upgrade their software and make sure everything keeps working?

That, and the fact that you said there was a lot more to come and yet nothing came at all;

> But that's just the beginning - we've got many more brilliant features up our sleeves that will make Electron.NET even better. Stay tuned!

And then again...

> We are still at the beginning and still have numerous performance and stability changes ahead of us. You can definitely stay curious how much power we have ready for you in the future..

And yet nothing comes, no information, no update. It's fine that's the case, but it does mean this project is no longer active.

> The focus is always where users are also supportive. Personally, when I see contributions/donations being made that is for me a sign that help is needed / appreciated; so I put the focus there.

And this confirms that basically the project will only be worked on if there is money to be made.

**Comment by FlorianRappl** [comment]
> And this confirms that basically the project will only be worked on if there is money to be made.

I think you got it wrong. The project will be worked on if its not a money loss. My day (as yours) only has 24 hours. If, instead of the work clients paid for, I work (for free) on Electron.NET - why would somebody give me something? I have 2 kids and a family. I need to sustain my living (as do you I guess).

There are a lot of projects based on Electron.NET - projects that put food on the table of the companies behind it. Yet, no one sponsors Electron.NET. So the situation seems to be fine for these companies(?!). If you really would want an update - you would either contribute the patches yourself or you would sponsor the project to put the prio on it.

There is a difference between just being a bit of a money-loss and essentially a big money loss. This project is not about getting rich (or where is that attitude coming from?!) but rather about having the time spent well funded / appreciated.

As long as the situation is like that I (and I think others) can only contribute when some time is available for whatever reason or if there is a long-hanging fruit to be taken. I did so in the past year and I will continue in that mode.

Bottom line: I would love to devote more of time to this (or some other .NET OSS) project. But truth is that everyone (incl. you) expects some slave to do all the work for free. This is not how it works.

**Remark**:
The funny thing is that you @nepp95 did not even star the project. You did not even make the minimum contribution needed to appreciate this project and yet you come and put a comment in here. Quite a sad story actually.

**Comment by scottkuhl** [comment]
The conversion here got pretty dark.  But through all the anger @nepp95  has a point.  The maintainers are not doing the minimum to support this project.  I know OSS is a commitment and can be a lot of work.   It's a shame you have not found a sponsor.  You are in a chicken and egg situation, where I can't propose to my company to support it until it shows activity and the project is not showing activity until it gets support.  We have moved on and it's too late to use this for our project.

If you go back to my comment on 12/3 you will see more upvotes than any other comment since the original two.  I proposed a solution to your problem:

> I recommended as a way to take this burden off the core team of maintainers guidelines be written on how to do it so anyone can pick it up.

But nothing came of it, not even a response from the maintainers.  You are not doing the minimum to make us feel like this project is active.  We can't build on a project that is running on old versions of .NET and Electron that are unsupported and you are not doing the bare minimum to make that happen.  Just accepting PRs from other people unless the project get sponsors is not enough.  That is not an active project or a sustainable attitude.

What I see is, you got excited to start working on the project again, made some promises, got bored, and did not keep them.   I'm sorry if this angers you, and its not meant to, but to show the reality of what is going on.

**Comment by nepp95** [comment]
> nepp95  I deeply despise your opinion. You were given a quality product for free, but you are unhappy.
> 
> What have you done for free for all of us? Funny and useless "EppoEngine"?
> 
> 
> 
> 

My opinion which is basically confronting someone with not following through on their commitments, while being censored and being insulted by one of the contributors because they don't like critique and the only responses I am getting aren't even answering any points I made. Yeah, we are making total sense here.

**Comment by FlorianRappl** [comment]
@scottkuhl you may have come up with a suggestion but what do you expect? I saw no one moving forward "hey I can do something". The truth is that no one seems to care.

> Also, I would consider issues, comments and discussions as contributions, not just PRs.

I never said contributions are limited to code. If you just take 5 min to investigate (listening to any podcast where I appeared on the topic) you would know that in contributions I always include such items.

> What I see is, you got excited to start working on the project again, made some promises, got bored, and did not keep them. I'm sorry if this angers you, and its not meant to, but to show the reality of what appears to be going on.

What angers me is that you make assumptions. I am not bored, but I have priorities. I did not make any promises - I came here to help and support, which I do / did. But my priority is on projects that have funding that I can live from it / justify doing that with my time. You cannot do that either. If you depend on Electron.NET then please start contributing in a meaningful way.

Otherwise we really need to shut this down (I, personally, would love to do the opposite - and invest more time in it; however, other projects are higher in my prio as people some to care more about them).

This will be my last post on the subject. I don't know how Gregor feels about the situation, but I think this is a pretty sad state of the community. I know that many other OSS maintainers also share that sentiment.

(Final remark: Actually I was thinking of devoting quite a substantial amount of time to Electron.NET later this year and talked about it with Gregor in February - but comments and discussions like that just de-motivate me to do so; why spend time on something for free where people are not even helpful and supportive but just destructive and arrogant? It may only be a few - but those seem to be the loudest, as usual - adding permanent damage.)

**Comment by scottkuhl** [comment]
@FlorianRappl I apologize if I angered you.  My intent was only to provide a perspective on appearances from an outside observer.

**Comment by r-pankevicius** [comment]
@jektvik We migrating our Electron.NET based project to [Photino.NET](https://github.com/tryphotino/photino.NET). Tried to play with Electron.NET fork [ElectronSharp](https://github.com/theolivenbaum/electron-sharp/), but it had it's own issues, too.
Now we're happy without electron, it was just unnecessary payload.
Of course the price we had to pay is creating the installer, set up won auto-update.

**Comment by dluc** [comment]
This would be a great project to keep alive, though it would be so much better if the frontend technology was decoupled from .NET. For instance, my preferred combo is React frontend + .NET backend. If there's any interest about going in that direction, please let me know. I know keeping OSS projects alive is a huge effort, and often dependent on personal passions and opinions :-)

**Comment by rajeshsuramalla** [comment]
> This would be a great project to keep alive, though it would be so much better if the frontend technology was decoupled from .NET. For instance, my preferred combo is React frontend + .NET backend. If there's any interest about going in that direction, please let me know. I know keeping OSS projects alive is a huge effort, and often dependent on personal passions and opinions :-)

This repo may be useful for you. it is an angular and dotnet with ElectronNet.

https://github.com/rajeshsuramalla/AngularWithDotNetCoreElectronNET
